webapp/tests/forms/flows/test_eligibility_step_chooser.py

```diff
@@ -75,3 +76,70 @@ def test_set_attributes_correctly(self):
             self.assertEqual(self.testing_steps, list(step_chooser.steps.values()))
             self.assertEqual(None, step_chooser.overview_step)
 
+
+class TestEligibilityStepChooserDeterminePrevStep(unittest.TestCase):
+    def setUp(self):
+        self.step_chooser = EligibilityStepChooser('eligibility')
+        self.step_chooser.step_order = []
+        self.step_chooser.steps = {}
+
+    def set_up_steps(self, steps_definitions):
+        for step_definition in steps_definitions:
+            step_name = step_definition['name']
+            mocked_step = MagicMock(is_previous_step=MagicMock(return_value=step_definition['is_previous_step']))
+            mocked_step.name = step_name
+            self.step_chooser.step_order.append(step_name)
+            self.step_chooser.steps[step_name] = mocked_step
+
+    def test_if_step_is_first_step_then_return_none(self):
+        self.set_up_steps([
+            {'name': 'step-0', 'is_previous_step': False},
+            {'name': 'step-1', 'is_previous_step': False},
+            {'name': 'step-2', 'is_previous_step': False}
+        ])
+
+        with patch('app.forms.flows.eligibility_step_chooser.get_session_data'):
+            prev_step = self.step_chooser.determine_prev_step('step-0')
+
+        self.assertIsNone(prev_step)
+
+    def test_if_no_step_returns_is_previous_step_true_then_return_first_in_list(self):
```

Comment:
This test looks the same as the one in l. 94 except for the given step name. So, maybe we could either rename those two a little to point out that difference or we add a comment to explain why we get None case and the next step in the other case.

Reply:
These cases are completely different? And I don't quite see how to make it more explicit. This here tests for *any* step, if none before returns True, then the first step ist returned. l. 94 tests only the *first* step that None is returned in that case.

Reply:
To make it more clear: if_given_step_is_first_step
or: variable with given_step_name
or: set False -> True in some cases for the first tests